Adds missing cauldron-item interactions.

emreozcan opened the pull request:
This PR adds the missing cauldron-item interactions listed in issue #1095:
- Cauldron &ndash; Empty bucket
- Cauldron &ndash; Water bottle

emreozcan committed:
Added missing cauldron-item interactions. Fixes GlowstoneMC#1095.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/GlowstoneMC/Glowstone?pullRequest=1097) <br/>All committers have signed the CLA.

mastercoms commented on src/main/java/net/glowstone/block/blocktype/BlockCauldron.java:
```diff
@@ -61,36 +75,74 @@ public boolean blockInteract(GlowPlayer player, GlowBlock block, BlockFace face,
+            return;
+        }
+        if (player.getGameMode() != GameMode.CREATIVE) {
+            Map<Integer, ItemStack> dummyInventory = player.getInventory()
```
Could we skip this if the bucket amount is 1, and just set the type to water bucket?

emreozcan replied:
Certainly. Doing the same thing to [Line 124](https://github.com/GlowstoneMC/Glowstone/pull/1097/files/c0bebd5ec90cbbcce286c10c667b053ddf1c98a5#diff-b80570db289b0cfacd9df3c246576452R124) too.

mastercoms commented on src/main/java/net/glowstone/block/blocktype/BlockCauldron.java:
```diff
@@ -61,36 +75,74 @@ public boolean blockInteract(GlowPlayer player, GlowBlock block, BlockFace face,
             }
         }
+
+        player.getItemInHand().setAmount(player.getItemInHand().getAmount() - 1);
```
Is the bucket still consumed in creative mode?

emreozcan replied:
Yes, oops. Will be fixed in next commit.

mastercoms commented on src/main/java/net/glowstone/block/blocktype/BlockCauldron.java:
```diff
@@ -61,36 +75,74 @@ public boolean blockInteract(GlowPlayer player, GlowBlock block, BlockFace face,
+            return;
+        }
+        if (player.getGameMode() != GameMode.CREATIVE) {
+            Map<Integer, ItemStack> dummyInventory = player.getInventory()
```
Can we rename the variable to `drops`, for consistency?

emreozcan replied:
Of course! Will comply in next commit.

emreozcan committed:
Comply with requested changes.

mastercoms commented on src/main/java/net/glowstone/block/blocktype/BlockCauldron.java:
```diff
@@ -36,11 +38,23 @@ public boolean blockInteract(GlowPlayer player, GlowBlock block, BlockFace face,
+                return true;
+
+            case POTION:
+                emptyBottle(player, block);
```
Change emptyBottle to return boolean then change this to

```java
if (emptyBottle(player, block) {
  return true;
}
break;
...
} // switch statement end
return super.blockInteract(player, block, face, clickedLoc);
```

Return false if PotionType != WATER.

emreozcan replied:
[Line 44](https://github.com/GlowstoneMC/Glowstone/pull/1097/files/c317b67b28b66bd0b06040f30cfe305cdb97a01b#diff-b80570db289b0cfacd9df3c246576452R44) already does this. Am I missing something?

mastercoms replied:
I was suggesting an alternative to that logic, which is more extensible for the future if we have more edge case checks we need to do for each item type.

emreozcan replied:
Okay, makes sense.

emreozcan committed:
Move potion edge case

Check if the player is holding a potion instead of a water bottle inside `emptyBottle`.
getType() only needs to be called once this way.

emreozcan commented:
The failed check is [caused by a file transfer/connectivity issue on Azure's end](https://dev.azure.com/glowstonemc/Glowstone/_build/results?buildId=525&view=logs&j=f93145ac-9471-53b7-dc09-8530b564757f&t=ad3b922e-7e52-5934-c164-59fdffd4c0f4&l=1438). Obviously I don't have access to retrying failed jobs but I hope someone who does can verify this isn't an issue with the PR.
